File: hsg9-tinh/2023-2024-cantho/chonqua/chonqua.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <numeric>

#define input_file "chonqua.inp"
#define output_file "chonqua.out"

using namespace std;

int n, k;
vector<int> A;
long long int Sum;


void input()
{
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    freopen(input_file, "r", stdin);

    cin >> n >> k;

    A.resize(n);

    for (int i = 0; i < n; ++i)
    {
        cin >> A[i];
    }
}


void process()
{
    // Sắp xếp mảng A theo thứ tự giảm dần
    sort(A.begin(), A.end(), greater<int>());

    // Tính tổng k phần tử đầu
    Sum = accumulate(A.begin(), A.begin() + k, 0);
}


void output()
{   
    freopen(output_file, "w", stdout);

    cout << Sum;
}


int main()
{
    input();
    process();
    output();

    return 0;
}

File: hsg9-tinh/2023-2024-cantho/chonqua/chonqua.py
import os
import sys

input_file = os.path.join(sys.path[0], 'chonqua.inp')
output_file = os.path.join(sys.path[0], 'chonqua.out')

n = k = 0
A = []
Sum = 0


def input():
    global n, k, A

    with open(input_file, 'r') as f:
        n, k = map(int, f.readline().split())

        A = list(map(int, f.readline().split()))


def process():
    global n, k, A, Sum

    # Sắp xếp mảng A theo thứ tự giảm dần
    A.sort(reverse=True)

    # Tính tổng k phần tử đầu
    Sum = sum(A[:k])


def output():
    global Sum

    with open(output_file, 'w') as f:
        f.write(str(Sum))


if __name__ == '__main__':
    input()
    process()
    output()